src/QsCompiler/CompilationManager/EditorSupport/CodeActions.cs

```diff
@@ -307,6 +307,33 @@ where DiagnosticTools.ErrorType(ErrorCode.UnknownType)(diagnostic)
             UnknownIdNamespaceSuggestions(file, compilation, lineNr, diagnostics)
                 .Concat(UnknownIdCaseSuggestions(file, compilation, diagnostics));
 
+        internal static IEnumerable<(string, WorkspaceEdit)> BindingSuggestions(
+            this FileContentManager file, CompilationUnit compilation, IReadOnlyCollection<Diagnostic> diagnostics)
+        {
+            var immutableBindingDiagnostics = diagnostics.Where(DiagnosticTools.ErrorType(ErrorCode.UpdateOfImmutableIdentifier));
+            if (immutableBindingDiagnostics.Any())
+            {
+                var edits = new List<(string, WorkspaceEdit)>();
+                foreach (var diagnostic in immutableBindingDiagnostics)
+                {
+                    var refs = file.TryGetReferences(compilation, diagnostic.Range.Start.ToQSharp(), out var declarationLocation, out _);
+
+                    if (refs && declarationLocation != null)
+                    {
+                        var overlappingFragments = file.FragmentsOverlappingWithRange(declarationLocation.Range.ToQSharp()).Where(f => f.Text != null && f.Text.TrimStart().StartsWith(Keywords.qsImmutableBinding.id));
```

Comment:
Very minor nitpick (feel free to ignore), for all the people with limited screen width:
```suggestion
                        var overlappingFragments = file
                            .FragmentsOverlappingWithRange(declarationLocation.Range.ToQSharp())
                            .Where(f => f.Text != null && f.Text.TrimStart()
                            .StartsWith(Keywords.qsImmutableBinding.id));
```

Reply:
makes sense!